plugins/providers/virtualbox/cap/cleanup_disks.rb

```diff
@@ -28,23 +29,52 @@ def self.handle_cleanup_disk(machine, defined_disks, disk_meta)
           vm_info = machine.provider.driver.show_vm_info
           primary_disk = vm_info["SATA Controller-ImageUUID-0-0"]
 
-          disk_meta.each do |d|
-            dsk = defined_disks.select { |dk| dk.name == d["name"] }
-            if !dsk.empty? || d["uuid"] == primary_disk
-              next
-            else
-              LOGGER.warn("Found disk not in Vagrantfile config: '#{d["name"]}'. Removing disk from guest #{machine.name}")
-              disk_info = machine.provider.driver.get_port_and_device(d["uuid"])
+          unless disk_meta.nil?
+            disk_meta.each do |d|
+              dsk = defined_disks.select { |dk| dk.name == d["name"] }
+              if !dsk.empty? || d["uuid"] == primary_disk
+                next
+              else
+                LOGGER.warn("Found disk not in Vagrantfile config: '#{d["name"]}'. Removing disk from guest #{machine.name}")
+                disk_info = machine.provider.driver.get_port_and_device(d["uuid"])
 
-              machine.ui.warn("Disk '#{d["name"]}' no longer exists in Vagrant config. Removing and closing medium from guest...", prefix: true)
+                machine.ui.warn("Disk '#{d["name"]}' no longer exists in Vagrant config. Removing and closing medium from guest...", prefix: true)
 
-              if disk_info.empty?
-                LOGGER.warn("Disk '#{d["name"]}' not attached to guest, but still exists.")
-              else
-                machine.provider.driver.remove_disk(disk_info[:port], disk_info[:device])
+                if disk_info.empty?
+                  LOGGER.warn("Disk '#{d["name"]}' not attached to guest, but still exists.")
+                else
+                  machine.provider.driver.remove_disk(disk_info[:port], disk_info[:device])
+                end
+
+                machine.provider.driver.close_medium(d["uuid"])
               end
+            end
+          end
+        end
 
-              machine.provider.driver.close_medium(d["uuid"])
+        # @param [Vagrant::Machine] machine
+        # @param [VagrantPlugins::Kernel_V2::VagrantConfigDisk] defined_dvds
+        # @param [Hash] dvd_meta - A hash of all the previously defined dvds from the last configure_disk action
+        def self.handle_cleanup_dvd(machine, defined_dvds, dvd_meta)
+          controller = "IDE Controller"
+          vm_info = machine.provider.driver.show_vm_info
+
+          unless dvd_meta.nil?
+            dvd_meta.each do |d|
+              dsk = defined_dvds.select { |dk| dk.name == d["name"] }
+              if !dsk.empty?
+                next
+              else
+                LOGGER.warn("Found dvd not in Vagrantfile config: '#{d["name"]}'. Removing dvd from guest #{machine.name}")
+                (0..1).each do |port|
+                  (0..1).each do |device|
+                    if d["uuid"] == vm_info["#{controller}-ImageUUID-#{port}-#{device}"]
```

Comment:
This is now possible (and implemented!) because I gave up on trying to support duplicate ISO attachments on a single guest. https://github.com/hashicorp/vagrant/pull/11613/files#diff-3494e15e76d59a15896961f458368a54R76